Model/Api/DiscountCodeValidation.php

```diff
@@ -452,6 +461,9 @@ public function validate()
                 return false;
             }
 
+            //remove previously cached Bolt order since we altered related immutable quote by applying a discount
+            $this->cache->clean([CartHelper::BOLT_ORDER_TAG . '_' . $parentQuoteId]);
```

Comment:
@danac-gs Does it mean we clear all the cache associated with the parent quote? if so, it seems to be aggressive somehow.

Reply:
@dumega 

All that is cached (and cleared) is the response from bolt (cart with order token). It may seem aggressive but it's cache after all, not essential for correct and stable functionality.

I still believe that this is overall the most optimal solution. We clear the cache when discount is applied. In vast majority of cases we won't be needing the old cached values after this step.

That said, it would be better (almost perfect) if we could tag the cache entry with the Bolt order token and remove only that single entry. Unfortunately, order token is not passed with the cart in the discount validation request.

Anyway, this change is something that won't be ever noticed in practice in terms of website execution speed. The best (and only) place for optimization here is making the code simple and clean.